File: 2016/src/util/input.ts
import { readFileSync } from 'fs'

export function getRawInput() {
  return readFileSync('input.txt', 'utf8')
}

export function getInput() {
  return getRawInput().trim()
}

export function getSplitInput(delimiter = '\n') {
  return getInput().split(delimiter)
}

export function getInputLines() {
  return getSplitInput().map(l => l.trim())
}

File: 2015/src/util/input.py
def get_input() -> str:
    with open("input.txt") as file:
        return file.read()


def get_input_lines() -> list[str]:
    with open("input.txt") as file:
        return [line.strip() for line in file.readlines()]


def get_split_input(delimiter: str) -> list[str]:
    with open("input.txt") as file:
        return file.read().split(delimiter)
